holidays/countries/brunei.py

```diff
@@ -0,0 +1,450 @@
+        # Status: In-Use.
+        # Starts in 1984.
+
+        _add_observed(self._add_holiday(tr("Hari Kebangsaan"), FEB, 23))
+
+        # Armed Forces Day
+        # Hari Angkatan Bersenjata Diraja Brunei
+        # Status: In-Use.
+        # Starts in 1984.
+        # Commemorates the formation of Royal Brunei Malay Regiment in 1961.
+
+        _add_observed(
+            self._add_holiday(
+                tr("Hari Angkatan Bersenjata Diraja Brunei"), MAY, 31
+            )
+        )
+
+        # Sultan Hassanal Bolkiah's Birthday
+        # Hari Keputeraan KDYMM Sultan Brunei
+        # Status: In-Use.
+        # Started in 1968.
+
+        _add_observed(
+            self._add_holiday(
+                tr("Hari Keputeraan KDYMM Sultan Brunei"), JUL, 15
+            )
+        )
+
+        # Christmas Day.
+        # Hari Natal
+        # Status: In-Use.
+
+        _add_observed(self._add_christmas_day(tr("Hari Natal")))
+
+        # Islamic Holidays are placed after Gregorian holidays to prevent
+        # the duplication of observed tags. - see #1168
+
+        # Isra Mi'raj
+        # Israk dan Mikraj
+        # Status: In-Use.
+
+        for dt in self._add_isra_and_miraj_day(tr("Israk dan Mikraj")):
+            _add_observed(dt)
+
+        # First Day of Ramadan
+        # Hari Pertama Berpuasa
+        # Status: In-Use.
+
+        for dt in self._add_ramadan_beginning_day(tr("Hari Pertama Berpuasa")):
+            _add_observed(dt)
+
+        # Anniversary of the revelation of the Quran
+        # Hari Nuzul Al-Quran
+        # Status: In-Use.
+
+        for dt in self._add_nuzul_al_quran_day(tr("Hari Nuzul Al-Quran")):
+            _add_observed(dt)
+
+        # Eid al-Fitr
+        # Hari Raya Aidil Fitri
+        # Status: In-Use.
+        # This is celebrate for three days in Brunei.
+        # Observed as 'Hari Raya Puasa' and only for 2 days at certain point.
+
+        name = tr("Hari Raya Aidil Fitri")
+
+        alfitr = self._add_eid_al_fitr_day(name)
+        self._add_eid_al_fitr_day_two(name)
+        self._add_eid_al_fitr_day_three(name)
+
+        # Eid al-Fitr (Observed)
+        # Hari Raya Aidil Fitri - Diperhatikan
+        # 1: If Wed-Thu-Fri -> Sat (+3)
+        # 2: If Thu-Fri-Sat -> Mon (+4)
+        # 3: If Fri-Sat-Sun -> Mon (+3)
+        # 4: If Sat-Sun-Mon -> Tue (+3)
+        # 5: If Sun-Mon-Tue -> Wed (+3)
+
+        aidil_fitri_in_lieu = self.tr("%s - Diperhatikan") % self.tr(
+            "Hari Raya Aidil Fitri"
+        )
+
+        if self.observed:
+            for dt in alfitr:
+                if (
+                    self._is_wednesday(dt)
+                    or self._is_friday(dt)
+                    or self._is_saturday(dt)
+                    or self._is_sunday(dt)
+                ):
+                    self._add_holiday(aidil_fitri_in_lieu, dt + td(days=+3))
+                elif self._is_thursday(dt):
+                    self._add_holiday(aidil_fitri_in_lieu, dt + td(days=+4))
```

Comment:
Something like that:
```suggestion
        if self.observed:
            for dt in alfitr:
                obs_date = None
                for delta in range(3):
                    hol_date = dt + td(days=delta)
                    if self._is_friday(hol_date) or self._is_sunday(hol_date):
                        obs_date = dt + td(days=+3)
                        if self._is_sunday(obs_date):
                            obs_date += td(days=+1)
                        break
                if obs_date:
                    self._add_holiday(
                        self.tr("%s - Diperhatikan") % self[hol_date], obs_date
                    )
```

Reply:
[BN] Implementing suggestions Part 2